prov/efa/src/efa_av.c

```diff
@@ -596,13 +659,14 @@ static struct fi_ops_av efa_av_ops = {
 static int efa_av_close(struct fid *fid)
 {
 	struct efa_av *av;
-	struct efa_av_entry *current_av_entry, *tmp;
 	int ret = 0;
 	int err = 0;
 	int i;
+	int ah_count;
 
 	av = container_of(fid, struct efa_av, util_av.av_fid.fid);
-	for (i = 0; i < av->util_av.count; i++) {
+	ah_count = MAX(av->used, av->util_av.count);
```

Comment:
Spoke with @dipti-kothari about this offline, there shouldn't be a case where av->used > av->util_av.count

Reply:
Yes, we are resizing when av->used equals av->util_av.count and setting av->util_av.count to the new value. Fixed